### PR TITLE
don't rebuild tables more than once + logging

### PR DESCRIPTION
**Pull request by czue**
still having a :imp: of a time chasing down http://manage.dimagi.com/default.asp?211297

The duplicate rebuilds was something I noticed while testing this. I don't quite see how it would lead to the behavior in that ticket, but is definitely better behavior 

@snopoke cc @proteusvacuum

### CODE REVIEW

File: corehq/apps/userreports/pillow.py [hunk 1]
```diff
@@ -70,19 +70,29 @@ def rebuild_tables_if_necessary(self):
                 migration_context = get_migration_context(connection, table_map.keys())
                 diffs = compare_metadata(migration_context, metadata)
 
-            tables_to_rebuild = get_tables_to_rebuild(diffs, table_map.keys())
+            tables_to_rebuild = set(get_tables_to_rebuild(diffs, table_map.keys()))
```
Comment (czue) [inline corehq/apps/userreports/pillow.py]: locally, this was rebuilding the same table 5 times when I made a change before adding this

**Comment by czue** [comment]
:white_check_mark:

File: corehq/apps/userreports/pillow.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/apps/userreports/pillow.py]: Is there something in the `get_tables_to_rebuild` method that is causing it to return the same thing 5 times?

File: corehq/apps/userreports/pillow.py [hunk 2]
```diff
@@ -70,19 +70,29 @@ def rebuild_tables_if_necessary(self):
...
-                    _notify_cory(u'rebuilt table {} ({}) because {}'.format(
-                        table_name,
-                        u'{} [{}]'.format(sql_adapter.config.display_name, sql_adapter.config._id),
-                        diffs,
-                    ))
+                    # note: this fancy logging can be removed as soon as we get to the
+                    # bottom of http://manage.dimagi.com/default.asp?211297
+                    # if no signs of it popping back up by april 2016, should remove this
+                    rev_after_rebuild = sql_adapter.config.get_db().get_rev(sql_adapter.config._id)
+                    _notify_cory(
```
Comment (proteusvacuum) [inline corehq/apps/userreports/pillow.py]: I :heart: that we have a `_notify_cory` function.

File: corehq/apps/userreports/pillow.py (hunk 1 above)
Comment (czue) [inline corehq/apps/userreports/pillow.py]: i didn't dig in to that too closely. @snopoke understands how that works, but basically I think `compare_metadata` returns multiple diffs for the same table and then they weren't being reconciled. I suspect something could be changed around [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/fluff/signals.py#L75) or we could just add the `set` call [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/fluff/signals.py#L79)

Comment (proteusvacuum) [inline corehq/apps/userreports/pillow.py]: cool. will defer to @snopoke if he thinks it is better to add the `set` higher up the chain, otherwise LGTM.

Comment (snopoke) [inline corehq/apps/userreports/pillow.py]: :100: to make `get_tables_to_rebuild` return a `set`

**Comment by czue** [comment]
alright, moved change and think this is good to go

File: corehq/apps/userreports/pillow.py (hunk 2 above)
Comment (benrudolph) [inline corehq/apps/userreports/pillow.py]: lol

**Pull request merged**